### PR TITLE
kms: add new samples and adopt new format

### PR DESCRIPTION
**Pull request by sethvargo**
This updates the Cloud KMS samples to use the new sample format, layout, and testing. Please review, but do not merge, because we have to coordinate a merge across the other languages.

### CODE REVIEW

**Comment by sethvargo** [comment]
Build seems to be failing due to unrelated Spanner tesets?

File: kms/create_key_asymmetric_decrypt.go [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// Copyright 2019 Google LLC
+// Copyright 2020 Google LLC
```
[the remark this answers is not in the document]
Comment (sethvargo) [reply kms/create_key_asymmetric_decrypt.go:1]: Yup 😄

File: kms/kms_helpers_test.go [hunk 2]
```diff
@@ -0,0 +1,293 @@
...
+	// Iterate over all keys and clean them up
+	keyIt := k.client.ListCryptoKeys(ctx, &kmspb.ListCryptoKeysRequest{
+		Parent: k.KeyRingName,
+	})
+	for {
+		key, err := keyIt.Next()
+		if err == iterator.Done {
+			break
+		}
+		if terr, ok := grpcstatus.FromError(err); ok && terr.Code() == grpccodes.NotFound {
```
[the remark this answers is not in the document]
Comment (sethvargo) [reply kms/kms_helpers_test.go:100]: TIL

File: kms/update_key_update_labels.go [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+	//
+	// Step 2 - add a label to the list of labels
+	//
+
+	labels := result.Labels
+	labels["new_label"] = "new_value"
+
+	// Build the request.
+	updateReq := &kmspb.UpdateCryptoKeyRequest{
```
[the remark this answers is not in the document]
Comment (sethvargo) [reply kms/update_key_update_labels.go:62]: Sadly there is not

File: kms/disable_key_version.go [hunk 4]
```diff
@@ -14,42 +14,46 @@
...
 import (
 	"context"
 	"fmt"
 	"io"
 
-	cloudkms "cloud.google.com/go/kms/apiv1"
+	kms "cloud.google.com/go/kms/apiv1"
 	kmspb "google.golang.org/genproto/googleapis/cloud/kms/v1"
-	fieldmask "google.golang.org/genproto/protobuf/field_mask"
+	"google.golang.org/genproto/protobuf/field_mask"
```
[the remark this answers is not in the document]
Comment (sethvargo) [reply kms/disable_key_version.go]: Updated. My line of thinking here was typing `field_mask.Thing` in a new project would automatically find the import using goimports and other tools, whereas the aliased package name would not.

File: kms/quickstart/quickstart.go [hunk 5]
```diff
@@ -22,43 +22,44 @@ import (
...
+
+	// Iterate and print the results.
 	for {
 		resp, err := it.Next()
 		if err == iterator.Done {
 			break
 		}
 		if err != nil {
-			log.Fatalf("Failed to list key rings: %v", err)
+			log.Fatalf("Failed to list KeyRings: %v", err)
```
Comment (tbpg) [inline kms/quickstart/quickstart.go:59]: "key rings"?

File: kms/quickstart/quickstart.go [hunk 6]
```diff
@@ -22,43 +22,44 @@ import (
...
 		}
-		fmt.Printf("KeyRing: %q\n", resp.Name)
+
+		fmt.Printf("KeyRing: %s\n", resp.Name)
```
Comment (tbpg) [inline kms/quickstart/quickstart.go:62]: "Key ring"?

File: go.mod [hunk 7]
```diff
@@ -22,7 +22,7 @@ require (
 	github.com/fluent/fluent-logger-golang v1.5.0
 	github.com/go-sql-driver/mysql v1.5.0
 	github.com/gofrs/uuid v3.2.0+incompatible
-	github.com/golang/protobuf v1.3.5
+	github.com/golang/protobuf v1.4.0
```
Comment (tbpg) [inline go.mod:25]: This upgrade is what is causing the Spanner tests to fail. Will you revert this (`go get github.com/golang/protobuf@v1.3.5` and `go mod tidy`)?

Comment (sethvargo) [reply go.mod]: Done

**Comment by tbpg** [comment]
Holding off on merging until the `do not merge` label is removed.

**Pull request merged**